toxcore/DHT.c

```diff
@@ -3031,3 +3034,49 @@ bool dht_non_lan_connected(const DHT *dht)
 
     return false;
 }
+
+/* Copies our own ip_port structure to dest. WAN addresses take priority over LAN addresses.
```

Comment:
In what situation do you want `dest` to be left alone? One problem with this function is that if `ipport_isset(dest)` is true before calling the function, this function potentially behaves differently from when it is false. Maybe we want to clear dest before starting the loop?